src/app/settings/options.component.ts

```diff
@@ -80,6 +90,16 @@ export class OptionsComponent implements OnInit {
         await this.stateService.save('enableGravatars', this.enableGravatars);
         await this.storageService.save('enableFullWidth', this.enableFullWidth);
         this.messagingService.send('setFullWidth');
+        if (this.theme !== this.startingTheme) {
+            this.changeTheme = this.startingTheme = this.theme;
+            await this.storageService.save('theme', this.changeTheme);
```

Comment:
Use `this.theme` as the source to save - it's all the same value at the moment, but it makes it clearer that you're saving the user's selection, and not the "effective" theme.